pkg/commands/image/upload/upload_olvm.go

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) 2024, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package upload
+
+import (
+	"fmt"
+	"github.com/docker/go-units"
+	"github.com/oracle-cne/ocne/pkg/cluster/driver/olvm"
+	otypes "github.com/oracle-cne/ocne/pkg/config/types"
+	"github.com/oracle-cne/ocne/pkg/file"
+	"github.com/oracle-cne/ocne/pkg/k8s/client"
+	"github.com/oracle-cne/ocne/pkg/ovirt/ovclient"
+	ovdisk "github.com/oracle-cne/ocne/pkg/ovirt/rest/disk"
+	ovit "github.com/oracle-cne/ocne/pkg/ovirt/rest/imagetransfer"
+	ovsd "github.com/oracle-cne/ocne/pkg/ovirt/rest/storagedomain"
+	"github.com/oracle-cne/ocne/pkg/util/logutils"
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// UploadOlvm uploads a boot image to an OLVM disk.
+// see https://ovirt.github.io/ovirt-imageio/images.html
+func UploadOlvm(o UploadOptions) error {
+	// get a kubernetes client
+	_, kubeClient, err := client.GetKubeClient(o.KubeConfigPath)
+	if err != nil {
+		return err
+	}
+
+	oCluster := o.ClusterConfig.Providers.Olvm.OlvmCluster
```

Comment:
Could this return null?

Reply:
no it is not a pointer, the providers are all inline values